packages/server/integrations/JiraServerTaskIntegrationManager.ts

```diff
@@ -44,15 +45,21 @@ export default class JiraServerTaskIntegrationManager implements TaskIntegration
     // TODO: implement stateToJiraServerFormat
     const description = contentState.getPlainText()
 
-    const res = await manager.createIssue(integrationRepoId, summary, description)
+    const {repoId, providerId} = IntegrationRepoId.split(integrationRepoId)
+
+    if (parseInt(providerId ?? '', 10) !== this.provider.id || !repoId) {
```

Comment:
-1 if `providerId` is supposed to be an int, `IntegrationRepoId.split` should already return an int